numpyro/contrib/reparam.py

```diff
@@ -55,6 +55,14 @@ def process_message(self, msg):
         new_fn, value = reparam(msg["name"], msg["fn"], msg["value"])
 
         if value is not None:
+            if new_fn is None:
+                msg['type'] = 'deterministic'
+                msg['value'] = value
+                for key in list(msg.keys()):
+                    if key not in ('type', 'name', 'value'):
```

Comment:
Could you explain why we need to delete here?

Reply:
We don't need to delete, but `deterministic` default msg doesn't hold those keys, so I delete for consistency.